Azure/sonic-buildimage
# Build sonic-ztp package

- Add changes in make rules to conditionally include sonic-ztp package

Signed-off-by: Rajendra Dendukuri <rajendra.dendukuri@broadcom.com>



**- What I did**
Added changes in makefiles to conditionally include sonic-ztp package. By default it is disabled.

**- How I did it**
- Created a config parameter for ZTP in rules/config. By default it is disabled.
- Modified slave.mk to include sonic-ztp deb file as a dependency for sonic-xxx.bin
- Modified sonic_debian_extension.j2 to install sonic-ztp deb in fsroot
- Modified Makefile.work to include a variable to indicate if ztp package is included in the build or not

**- How to verify it**
Modify rules/config and set ENABLE_ZTP=y and then build sonic installer bin. Notice that sonic-ztp deb file is created in target/debs/stretch/sonic-ztp-xxxx.deb

**- Description for the changelog**
Include make recipes for sonic-ztp package



**- A picture of a cute animal (not mandatory but encouraged)**
